celer/homotopy.py

```diff
@@ -325,6 +310,58 @@ def celer_path(X, y, pb, eps=1e-3, n_alphas=100, alphas=None,
     return results
 
 
+def _check_pb(pb, X, y, groups):
+    """Handle pb cases."""
+    # vars
+    pb = pb.lower()
+    grp_ptr, grp_indices, n_groups = None, None, None
+
+    # type of the problem
+    if pb not in ("lasso", "logreg", "grouplasso"):
+        raise ValueError("Unsupported problem %s" % pb)
+    if pb == "lasso":
+        pb = LASSO
+    elif pb == "logreg":
+        pb = LOGREG
+        if set(y) - set([-1.0, 1.0]):
+            raise ValueError(
+                "y must contain only -1. or 1 values. Got %s " % (set(y))
+            )
+    elif pb == "grouplasso":
+        pb = GRPLASSO
+        if groups is None:
+            raise ValueError(
+                "Groups must be specified for the group lasso problem."
+            )
+        grp_ptr, grp_indices = _grp_converter(groups, X.shape[1])
+        n_groups = len(grp_ptr) - 1
+    else:
+        raise ValueError("Unsupported problem: %s" % pb)
+
+    return pb, grp_ptr, grp_indices, n_groups
+
+
+def _check_weights(weights, pb, X, n_groups):
+    """Handle weights cases."""
+    if weights is None:
+        n_weights = n_groups if pb == GRPLASSO else X.shape[1]
+
+        weights = np.ones(n_weights).astype(X.dtype)
+    # weights are provided
+    elif (weights <= 0).any():
+        raise ValueError("0 or negative weights are not supported.")
+    else:
+        expected_n_weights = n_groups if pb == GRPLASSO else X.shape[1]
+
+        if weights.shape[0] != expected_n_weights:
+            raise ValueError(
+                "As many weights as groups must be passed. "
```

Comment:
you raise an error message with "groups" in the case of the Lasso, this will confuse the user